### PR TITLE
Update C# client features

### PR DESCRIPTION
**Pull request by cwe1ss**
The new v0.0.10 release of the C# client now supports almost the same features as Java.

I couldn't run this locally so I couldn't test it. Hope it's fine though.

(resolves https://github.com/jaegertracing/jaeger-client-csharp/issues/51)

### CODE REVIEW

**Comment by yurishkuro** [comment]
Tip: if you can't test locally you can check the preview in the PR

File: data/clients.yaml [hunk 1]
```diff
@@ -291,38 +291,38 @@ clients:
...
-      env_JAEGER_REPORTER_FLUSH_INTERVAL: no
-      env_JAEGER_SAMPLER_TYPE: no
-      env_JAEGER_SAMPLER_PARAM: no
-      env_JAEGER_SAMPLER_MANAGER_HOST_PORT: no
-      env_TRACER_TAGS: no
+      configuration: yes
+      config_tracer_tags: yes
+      remote_sampler: yes
+      adaptive_sampler: yes
+      baggage_restriction: coming
```
Comment (Falco20019) [inline data/clients.yaml]: Restrictions are completely implemented now, right?

Comment (Falco20019) [reply data/clients.yaml]: @cwe1ss Friendly ping

**Comment by yurishkuro** [comment]
I a going to merge this, minor corrections can be done separately. Baggage restrictions are not well supported from the backend side anyway - @black-adder is this something we can add as a static config?

**Pull request merged**